### PR TITLE
Only allow async restore with openrosa version 2.0

### PR DESCRIPTION
**Pull request by proteusvacuum**
In 2.29 the OpenRosa version is bumped to 2.0. This is the only version of CommCare that allows async restores.

Also added that this can be sent as a GET param so we can still test things in the admin restore view (need to set `&openrosa_version=2.0`)

@dannyroberts @amstone326

### CODE REVIEW

File: corehq/apps/ota/views.py [hunk 1]
```diff
@@ -140,7 +153,7 @@ def get_restore_response(domain, couch_user, app_id=None, since=None, version='1
 
     project = Domain.get_by_name(domain)
     app = get_app(domain, app_id) if app_id else None
-    async_restore = toggles.ASYNC_RESTORE.enabled(domain)
+    async_restore = toggles.ASYNC_RESTORE.enabled(domain) and float(openrosa_version) >= OPENROSA_VERSION_MAP['ASYNC_RESTORE']
```
Comment (dimagimon) [inline corehq/apps/ota/views.py]: E501 line too long (126 > 115 characters)

**Comment by amstone326** [comment]
@proteusvacuum Let me know when this is merged and I then can test to make sure it's working

File: corehq/apps/ota/views.py [hunk 2]
```diff
@@ -113,13 +122,17 @@ def get_restore_params(request):
...
 
 
 def get_restore_response(domain, couch_user, app_id=None, since=None, version='1.0',
                          state=None, items=False, force_cache=False,
                          cache_timeout=None, overwrite_cache=False,
-                         force_restore_mode=None, as_user=None, has_data_cleanup_privelege=False):
+                         force_restore_mode=None,
+                         as_user=None,
+                         has_data_cleanup_privelege=False,
+                         openrosa_version=settings.OPENROSA_VERSION):
```
Comment (czue) [inline corehq/apps/ota/views.py]: not your fault but it seemed weird to me that this was in `settings`

File: corehq/apps/ota/views.py [hunk 3]
```diff
@@ -140,7 +153,8 @@ def get_restore_response(domain, couch_user, app_id=None, since=None, version='1
 
     project = Domain.get_by_name(domain)
     app = get_app(domain, app_id) if app_id else None
-    async_restore = toggles.ASYNC_RESTORE.enabled(domain)
+    async_restore = (toggles.ASYNC_RESTORE.enabled(domain) and
+                     float(openrosa_version) >= OPENROSA_VERSION_MAP['ASYNC_RESTORE'])
```
Comment (czue) [inline corehq/apps/ota/views.py]: nit: might make more sense to do with [`distutils`](http://stackoverflow.com/a/6972866/8207)

File: corehq/apps/ota/views.py (hunk 2 above)
Comment (proteusvacuum) [inline corehq/apps/ota/views.py]: Yeah, to me too... Should I just add it as a constant and set the default to be "1.0"?

File: corehq/apps/ota/views.py (hunk 3 above)
Comment (proteusvacuum) [inline corehq/apps/ota/views.py]: Very cool. Didn't know about that. Will update.

File: corehq/apps/ota/views.py (hunk 2 above)
Comment (czue) [inline corehq/apps/ota/views.py]: sure, that sounds good to me unless there's some reason you uncover that it's there

**Pull request merged**